Lushu/src/main/kotlin/lushu/ContextGrammar/Grammar/GrammarNode.kt

```diff
@@ -0,0 +1,204 @@
+package lushu.ContextGrammar.Grammar
+
+import lushu.ContextGrammar.Grammar.Nodes.NonMergeableNode
+import lushu.ContextGrammar.Grammar.Nodes.PlusNode
+import lushu.ContextGrammar.Grammar.Nodes.RegexNode
+import lushu.ContextGrammar.Grammar.Nodes.SensitiveNode
+
+class GrammarNode(
+    override var regex: String = "",
+    override var sensitive: Boolean = false,
+    override var plus: Boolean = false,
+    override var nonmergeable: Boolean = false,
+
+    private val children: MutableList<GrammarNode> = mutableListOf()
+
+) : RegexNode, SensitiveNode, PlusNode, NonMergeableNode {
```

Comment:
@Vitor-Git15 I think we can just add these flags in the nodes in the `Lattice` module... This is becoming way too complex, and we're losing time. Let's just make it work in the easiest way possible

I know we were planning to use the lattice product mathematical property to our benefit, but there is an edge case for the regex lattice that would need to be separately treated anyway.

Reply:
Just add these flags to the `Merger.Lattice.Node.Node`, please.